rest/models/request.go

```diff
@@ -20,15 +20,15 @@ type RequestOptions struct {
 // RequestOption changes the configuration of RequestOptions.
 type RequestOption func(o *RequestOptions)
 
-// WithAPIKey sets the APIKey for an Option.
-func WithAPIKey(id string) RequestOption {
+// APIKey sets an APIKey as an ption.
+func APIKey(id string) RequestOption {
```

Comment:
this is sort of a builder pattern but it doesn't appear that way to users so With felt somewhat confusing